src/test/java/com/zendesk/maxwell/row/RowMapTest.java

```diff
@@ -55,4 +56,198 @@ public void testTimestampConversion() throws Exception {
+
+		RowMap rowMap = new RowMap("insert", "MyDatabase", "MyTable", timestampMillis, pKeys, position);
+
+		rowMap.putData("id", "9001");
+		rowMap.putData("first_name", "foo");
+		rowMap.putData("middle_name", "buzz");
+		rowMap.putData("last_name", "bar");
+		rowMap.putData("salary", "4000");
+		rowMap.putData("department", "science");
+
+		List<String> partitionColumns = new ArrayList<>();
+
+		String partitionKey = rowMap.buildPartitionKey(partitionColumns, "primary_key");
+
+		Assert.assertEquals("9001foo", partitionKey);
+
+	}
+
+	@Test
+	public void testToJSONWithRawJSONData() throws IOException {
+		long timestampMillis = 1496712943487L;
+
+		Position position = new Position(new BinlogPosition(1L, "binlog-0001"), 0L);
+		RowMap rowMap = new RowMap("insert", "MyDatabase", "MyTable", timestampMillis,
+				null, position);
+
+		rowMap.putData("id", "9001");
+		rowMap.putData("first_name", "foo");
+		rowMap.putData("middle_name", "buzz");
+		rowMap.putData("last_name", "bar");
+		rowMap.putData("salary", "4000");
+		rowMap.putData("department", "science");
+        rowMap.putData("rawJSON", new RawJSONString("{\"database\":\"MyDatabase\",\"table\":\"User\",\"type\":\"insert\",\"ts\":1486439516,\"xid\":5694,\"commit\":true,\"data\":{\"UserID\":20,\"FirstName\":\"Fiz\",\"MiddleName\":null,\"LastName\":\"Buz\",\"Version\":1486703131000}}\n"));
+		MaxwellOutputConfig outputConfig = getMaxwellOutputConfig();
+
+		Assert.assertEquals("{\"database\":\"MyDatabase\",\"table\":\"MyTable\",\"type\":\"insert\",\"ts\":1496712943,\"position\":\"binlog-0001:1\",\"gtid\":null,\"data\":{\"id\":\"9001\",\"salary\":\"4000\",\"department\":\"science\",\"rawJSON\":{\"database\":\"MyDatabase\",\"table\":\"User\",\"type\":\"insert\",\"ts\":1486439516,\"xid\":5694,\"commit\":true,\"data\":{\"UserID\":20,\"FirstName\":\"Fiz\",\"MiddleName\":null,\"LastName\":\"Buz\",\"Version\":1486703131000}}\n" +
+				"}}", rowMap.toJSON(outputConfig));
+	}
+
+	@Test
+	public void testToJSONWithListData() throws IOException {
+		long timestampMillis = 1496712943487L;
+
+		Position position = new Position(new BinlogPosition(1L, "binlog-0001"), 0L);
+		RowMap rowMap = new RowMap("insert", "MyDatabase", "MyTable", timestampMillis,
+				null, position);
+
+		rowMap.putData("id", "9001");
+		rowMap.putData("first_name", "foo");
+		rowMap.putData("middle_name", "buzz");
+		rowMap.putData("last_name", "bar");
+		rowMap.putData("salary", "4000");
+		rowMap.putData("department", "science");
+		rowMap.putData("interests", Arrays.asList("classical music", "hiking", "biking", "programming"));
+		MaxwellOutputConfig outputConfig = getMaxwellOutputConfig();
+
+		Assert.assertEquals("{\"database\":\"MyDatabase\",\"table\":\"MyTable\",\"type\":\"insert\",\"ts\":1496712943,\"position\":\"binlog-0001:1\",\"gtid\":null,\"data\":{\"id\":\"9001\",\"salary\":\"4000\",\"department\":\"science\",\"interests\":[\"classical music\",\"hiking\",\"biking\",\"programming\"]}}", rowMap.toJSON(outputConfig));
```

Comment:
Again, we probably don't need this much sample data - if it works with one & two fields, chances are it's going to work with 3+.